### PR TITLE
percy(chore): 1739 percy integration and storybook cleanup

### LINKED ISSUE #1739: Percy integration and storybook cleanup
Let's get Percy running again, in some shape or form, as part of our process.

### PR DESCRIPTION
**Pull request by ArtBlue**
Fixes #1739 

- [ ] This PR contains CSS changes
- [x] This PR does not contain CSS changes

## Description
Upgraded Percy CLI, added dedicated scripts for running snapshots for all stories as well as specific individual stories. I also cleaned out a couple of issues I found with stories.

## Notes
1. There’s no good way to exclude stories from snapshot exclusion once stories are listed in `.percy.yml`. It can be done but adding back inclusions for snapshots to make it additive and does not replace the `include[]` list in `.percy.yml`.
1. The best approach to doing snapshots most flexibly via CLI is to not include any story in the `.percy.yml` and run snapshots with either ALL or only select stories via regex with the  `--include` CLI arg.
1. More research will be needed to see if it’s possible to run snapshots selectively on github based on which components were modified in PR merges. At the moment, it does not appear to be possible.
1. Using an –include with a regex to run certain tests looks to be the best alternative:
`npm run snapshots:dry 'Button/Button/Primary'`

## Usage Notes

Snapshots will likely be ran by developers locally before pushing up changes. Once they are ran, Percy dashboard should be checked to ensure no unintended style changes have taken place. If there are intended style changes that have occurred, those should be reverted/fixed. Once the set of new local changes is final and in scope with the changes related to the issue, the snapshot run will need to be marked as the canonical version against which future updates are compared.

### Build Modes

Running snapshots has two modes for all variations - build mode and dry mode.

**Build mode** creates snapshots for stories, uploads the snapshots to Percy and created a new Percy build for the purposes of making comparisons to previous visual builds. 

**Dry mode** runs through snapshots (mainly for the purpose of verifying which stories will be captured) but does not upload the snapshots and does not create a Percy build.

### Run All Snapshots
Build mode: `npm run snapshots:all`
Dry mode: `npm run snapshots:all:dry` 

### Run Specific Snapshot(s)
Single build mode: `npm run snapshots 'Button/Button/Primary'`
Multiple build mode: `npm run snapshots 'Button/Button/Primary,Icon/'`

Single dry mode: `npm run snapshots:dry 'Button/Button/Primary'`
Multiple dry mode: `npm run snapshots:dry 'Button/Button/Primary,Icon/'`

## Checklist

- [x] I verify the build is in a non-broken state
- [x] I verify all changes are within scope of the linked issue

- [x] I regenerated all CSS files under dist folder
- [x] I tested the UI in all supported browsers
- [x] I tested the UI in dark mode and RTL mode
- [x] I added/updated/removed Storybook coverage as appropriate

### CODE REVIEW

**Comment by ArtBlue** [comment]
> Is there a way to do it without requiring regex in the command line each time? Can we leverage environment variables instead, like we are doing in the compileModule script?

Good idea. Should be doable I think. Assuming we move the regex portion of the input into `.gulpfile`, how would we want the user input format in the `CLI`?

Would `package.json` includes something like this?:
`"snapshot": "npx gulp runSnapshot --snapshots $SNAPSHOTS",`

...and in `.gulpfile` something like this?:

```
async function runSnapshot(snapshots) {
    percy storybook ./docs/storybook --include  '\b(?:' + snapshots + ')\b'
}
```

**Comment by ianmcburnie** [comment]
I wasn't actually thinking about using Gulp, but yeah that might make things easier. Initially I was just wondering if we could have it all self contained in the NPM script itself:

NPM Script:

`"snapshot": "percy storybook ./docs/storybook --include '\b(?:' + $STORY + ')\b'"`

To execute:

`$STORY=badge npm run snapshot`

But not sure if an NPM script can accommodate the regex syntax that gets passed to Percy. If not, then Gulp could be the way to go.

**Comment by ArtBlue** [comment]
> I wasn't actually thinking about using Gulp, but yeah that might make things easier. Initially I was just wondering if we could have it all self contained in the NPM script itself:
> 
> NPM Script:
> 
> `"snapshot": "percy storybook ./docs/storybook --include '\b(?:' + $STORY + ')\b'"`
> 
> To execute:
> 
> `$STORY=badge npm run snapshot`
> 
> But not sure if an NPM script can accommodate the regex syntax that gets passed to Percy. If not, then Gulp could be the way to go.

**Comment by ArtBlue** [comment]
> I wasn't actually thinking about using Gulp, but yeah that might make things easier. Initially I was just wondering if we could have it all self contained in the NPM script itself:
> 
> NPM Script:
> 
> `"snapshot": "percy storybook ./docs/storybook --include '\b(?:' + $STORY + ')\b'"`
> 
> To execute:
> 
> `$STORY=badge npm run snapshot`
> 
> But not sure if an NPM script can accommodate the regex syntax that gets passed to Percy. If not, then Gulp could be the way to go.

No, I tried that before (one of many things I tried) and I couldn't get that to work. It seems like `percy storybook --include ` expects the regex and does not do any sort of detection of what's being passed in to reformat the input into its usable form. 

Also, the CLI command format has to be flexible enough to accommodate more than one story snapshot. I can see instances where a change may impact a few stories and we would want to pass in multiple stories to run snapshots for. That sort of format would probably not work too well. 

I was hoping to have the script being run be as simple as possible. Given that, going with gulp is probably the best option. At the moment, the Percy CLI is not very flexible at all.

**Comment by ianmcburnie** [comment]
Aha - I see. Let's go with Gulp then.

**Comment by ArtBlue** [comment]
It looks like program badge is missing in storybook. Should it have stories or is it exempt like some of the other ones?

**Comment by agliga** [comment]
> It looks like program badge is missing in storybook. Should it have stories or is it exempt like some of the other ones?

Not sure. But lets leave it for now if we can.

**Review by ianmcburnie — CHANGES_REQUESTED** [review]
I think the regex might need adjusting. Here is the output from running it on "Button":

```
> @ebay/skin@13.7.1 snapshots:dry
> npx gulp runSnapshots --dry yes --stories $STORIES "Button"

[15:30:07] Using gulpfile ~/code/github/ebay/skin/gulpfile.js
[15:30:07] Starting 'runSnapshots'...

************************************************************
Running Percy Snapshot(s)...
Snapshot(s):         Button
Percy Stories Regex: \b(?:Button)\b
Percy Dry Run:       yes
Running Snapshot(s)...
************************************************************
	
[15:30:07] Finished 'runSnapshots' after 14 ms

> @ebay/skin@13.7.1 snapshots:execute:dry
> percy storybook ./docs/storybook --dry-run --include "\b(?:Button)\b"

[percy] Percy has started!
[percy] Snapshot found: Button/Button/Base: Busy
[percy] Snapshot found: Button/Button/Base: Busy Large
[percy] Snapshot found: Button/Button/Base: Fixed Width
[percy] Snapshot found: Button/Button/Base: Fluid
[percy] Snapshot found: Button/Button/Base: Form Reset
[percy] Snapshot found: Button/Button/Base: Form Submit
[percy] Snapshot found: Button/Button/Base: Icon And Text
[percy] Snapshot found: Button/Button/Base: Large
[percy] Snapshot found: Button/Button/Base: Text Only
[percy] Snapshot found: Button/Button/Base: Wide
[percy] Snapshot found: Button/Button/Base/Cascade: RTL
[percy] Snapshot found: Button/Button/Base/Cascade: Color
[percy] Snapshot found: Button/Button/Base/Cascade: Font Size
[percy] Snapshot found: Button/Button/Base/Fixed Height: Non Truncated
[percy] Snapshot found: Button/Button/Base/Fixed Height: Truncated
[percy] Snapshot found: Button/Button/Deprecated: Loading Base
[percy] Snapshot found: Button/Button/Deprecated: Loading Base Large
[percy] Snapshot found: Button/Button/Deprecated: Loading Destructive Primary
[percy] Snapshot found: Button/Button/Deprecated: Loading Destructive Secondary
[percy] Snapshot found: Button/Button/Deprecated: Loading Destructive Tertiary
[percy] Snapshot found: Button/Button/Deprecated: Loading Primary
[percy] Snapshot found: Button/Button/Deprecated: Loading Secondary
[percy] Snapshot found: Button/Button/Deprecated: Loading Tertiary
[percy] Snapshot found: Button/Button/Primary: Busy
[percy] Snapshot found: Button/Button/Primary: Disabled
[percy] Snapshot found: Button/Button/Primary: Icon And Text
[percy] Snapshot found: Button/Button/Primary: Icon Only
[percy] Snapshot found: Button/Button/Primary: Loading Deprecated
[percy] Snapshot found: Button/Button/Primary: Partially Disabled
[percy] Snapshot found: Button/Button/Primary: Text Only
[percy] Snapshot found: Button/Button/Secondary: Busy
[percy] Snapshot found: Button/Button/Secondary: Disabled
[percy] Snapshot found: Button/Button/Secondary: Icon And Text
[percy] Snapshot found: Button/Button/Secondary: Icon Only
[percy] Snapshot found: Button/Button/Secondary: Partially Disabled
[percy] Snapshot found: Button/Button/Secondary: Text Only
[percy] Snapshot found: Button/Button/Tertiary: Busy
[percy] Snapshot found: Button/Button/Tertiary: Disabled
[percy] Snapshot found: Button/Button/Tertiary: Tertiary
[percy] Snapshot found: Button/Button/Destructive Button/Primary: Busy
[percy] Snapshot found: Button/Button/Destructive Button/Primary: Disabled
[percy] Snapshot found: Button/Button/Destructive Button/Primary: Disabled With Icon
[percy] Snapshot found: Button/Button/Destructive Button/Primary: Icon And Text
[percy] Snapshot found: Button/Button/Destructive Button/Primary: Icon Only
[percy] Snapshot found: Button/Button/Destructive Button/Primary: Text Only
[percy] Snapshot found: Button/Button/Destructive Button/Secondary: Busy
[percy] Snapshot found: Button/Button/Destructive Button/Secondary: Disabled
[percy] Snapshot found: Button/Button/Destructive Button/Secondary: Disabled With Icon
[percy] Snapshot found: Button/Button/Destructive Button/Secondary: Icon And Text
[percy] Snapshot found: Button/Button/Destructive Button/Secondary: Icon Only
[percy] Snapshot found: Button/Button/Destructive Button/Secondary: Text Only
[percy] Snapshot found: Button/Button/Destructive Button/Tertiary: Busy
[percy] Snapshot found: Button/Button/Destructive Button/Tertiary: Disabled
[percy] Snapshot found: Button/Button/Destructive Button/Tertiary: Disabled With Icon
[percy] Snapshot found: Button/Button/Destructive Button/Tertiary: Icon And Text
[percy] Snapshot found: Button/Button/Destructive Button/Tertiary: Icon Only
[percy] Snapshot found: Button/Button/Destructive Button/Tertiary: Tertiary
[percy] Snapshot found: Button/Button/Destructive Button/Tertiary: Text Only
[percy] Snapshot found: Fake Button/Fake Button/Base: RTL
[percy] Snapshot found: Fake Button/Fake Button/Base: Fixed Width
[percy] Snapshot found: Fake Button/Fake Button/Base: Fluid
[percy] Snapshot found: Fake Button/Fake Button/Base: Icon And Text
[percy] Snapshot found: Fake Button/Fake Button/Base: Large
[percy] Snapshot found: Fake Button/Fake Button/Base: Text Only
[percy] Snapshot found: Fake Button/Fake Button/Base/Cascade: Color
[percy] Snapshot found: Fake Button/Fake Button/Base/Cascade: Font Size
[percy] Snapshot found: Fake Button/Fake Button/Base/Fixed Height: Non Truncated
[percy] Snapshot found: Fake Button/Fake Button/Base/Fixed Height: Truncated
[percy] Snapshot found: Fake Button/Fake Button/Primary: Disabled
[percy] Snapshot found: Fake Button/Fake Button/Primary: Icon And Text
[percy] Snapshot found: Fake Button/Fake Button/Primary: Icon Only
[percy] Snapshot found: Fake Button/Fake Button/Primary: Partially Disabled
[percy] Snapshot found: Fake Button/Fake Button/Primary: Text Only
[percy] Snapshot found: Fake Button/Fake Button/Secondary: Disabled
[percy] Snapshot found: Fake Button/Fake Button/Secondary: Icon And Text
[percy] Snapshot found: Fake Button/Fake Button/Secondary: Icon Only
[percy] Snapshot found: Fake Button/Fake Button/Secondary: Partially Disabled
[percy] Snapshot found: Fake Button/Fake Button/Secondary: Text Only
[percy] Snapshot found: CTA Button/CTA Button/Cascade: RTL
[percy] Snapshot found: CTA Button/CTA Button/Cascade: Color
[percy] Snapshot found: CTA Button/CTA Button/Cascade: Font Size
[percy] Snapshot found: CTA Button/CTA Button: Base
[percy] Snapshot found: CTA Button/CTA Button: Wide
[percy] Snapshot found: Expand Button/Expand Button/Cascade: RTL
[percy] Snapshot found: Expand Button/Expand Button/Cascade: Centered RTL
[percy] Snapshot found: Expand Button/Expand Button/Cascade: Color
[percy] Snapshot found: Expand Button/Expand Button/Cascade: Icon And Text RTL
[percy] Snapshot found: Expand Button/Expand Button/Deprecated/Primary: Active
[percy] Snapshot found: Expand Button/Expand Button/Deprecated/Primary: Disabled
[percy] Snapshot found: Expand Button/Expand Button/Deprecated/Secondary: Active
[percy] Snapshot found: Expand Button/Expand Button/Deprecated/Secondary: Disabled
[percy] Snapshot found: Expand Button/Expand Button: Collapsed
[percy] Snapshot found: Expand Button/Expand Button: Disabled
[percy] Snapshot found: Expand Button/Expand Button: Expanded
[percy] Snapshot found: Expand Button/Expand Button: Fixed Width
[percy] Snapshot found: Expand Button/Expand Button: Large
[percy] Snapshot found: Expand Button/Expand Button: Long Text
[percy] Snapshot found: Expand Button/Expand Button: Wide
[percy] Snapshot found: Expand Button/Expand Button/Fixed Height: Non Truncated
[percy] Snapshot found: Expand Button/Expand Button/Fixed Height: Truncated
[percy] Snapshot found: Filter/Filter/Button: Base
[percy] Snapshot found: Filter/Filter/Button: Disabled
[percy] Snapshot found: Filter/Filter/Button: Pressed
[percy] Snapshot found: Filter/Filter/Button: Truncated
[percy] Snapshot found: Filter/Filter/Button/Group: RTL
[percy] Snapshot found: Filter/Filter/Button/Group: Group
[percy] Snapshot found: Filter/Filter/Button/Link: Base
[percy] Snapshot found: Filter/Filter/Button/Link: Disabled
[percy] Snapshot found: Filter/Filter/Button/Link: Selected
[percy] Snapshot found: Filter/Filter/Button/Link: Truncated
[percy] Snapshot found: Filter/Filter/Menu Button: RTL
[percy] Snapshot found: Filter/Filter/Menu Button: Collapsed
[percy] Snapshot found: Filter/Filter/Menu Button: Disabled
[percy] Snapshot found: Filter/Filter/Menu Button: Expanded
[percy] Snapshot found: Filter/Filter/Menu Button: Overflow
[percy] Snapshot found: Filter/Filter/Menu Button: Pressed
[percy] Snapshot found: Filter/Filter/Menu Button: Truncated
[percy] Snapshot found: Filter/Filter/Menu: With Footer Button
[percy] Snapshot found: Fullscreen Dialog/Fullscreen Dialog: Back Button
[percy] Snapshot found: Fullscreen Dialog/Fullscreen Dialog: Close Button
[percy] Snapshot found: Fullscreen Dialog/Fullscreen Dialog: Secondary Button
[percy] Snapshot found: Icon Button/Icon Button/Alignment: Close
[percy] Snapshot found: Icon Button/Icon Button/Alignment: Menu
[percy] Snapshot found: Icon Button/Icon Button/Alignment: Settings
[percy] Snapshot found: Icon Button/Icon Button/Button: RTL Menu
[percy] Snapshot found: Icon Button/Icon Button/Button: Cart
[percy] Snapshot found: Icon Button/Icon Button/Button: Disabled Menu
[percy] Snapshot found: Icon Button/Icon Button/Button: Disabled Settings
[percy] Snapshot found: Icon Button/Icon Button/Button: Menu
[percy] Snapshot found: Icon Button/Icon Button/Button: Partially Disabled Menu
[percy] Snapshot found: Icon Button/Icon Button/Button: Partially Disabled Settings
[percy] Snapshot found: Icon Button/Icon Button/Button: Save
[percy] Snapshot found: Icon Button/Icon Button/Button: Settings
[percy] Snapshot found: Icon Button/Icon Button/Button/Badged: One Digit
[percy] Snapshot found: Icon Button/Icon Button/Button/Badged: Three Digits
[percy] Snapshot found: Icon Button/Icon Button/Button/Badged: Two Digits
[percy] Snapshot found: Icon Button/Icon Button/Link: Disabled Menu
[percy] Snapshot found: Icon Button/Icon Button/Link: Disabled Settings
[percy] Snapshot found: Icon Button/Icon Button/Link: Menu
[percy] Snapshot found: Icon Button/Icon Button/Link: Partially Disabled Menu
[percy] Snapshot found: Icon Button/Icon Button/Link: Partially Disabled Settings
[percy] Snapshot found: Icon Button/Icon Button/Link: Settings
[percy] Snapshot found: Icon Button/Icon Button/Link/Badged: One Digit
[percy] Snapshot found: Icon Button/Icon Button/Link/Badged: Three Digits
[percy] Snapshot found: Icon Button/Icon Button/Link/Badged: Two Digits
[percy] Snapshot found: Listbox Button/Listbox Button/Base: Collapsed Selected
[percy] Snapshot found: Listbox Button/Listbox Button/Base: Collapsed Unselected
[percy] Snapshot found: Listbox Button/Listbox Button/Base: Disabled
[percy] Snapshot found: Listbox Button/Listbox Button/Base: Expanded Selected
[percy] Snapshot found: Listbox Button/Listbox Button/Base: Expanded Unselected
[percy] Snapshot found: Listbox Button/Listbox Button/Base: Invalid
[percy] Snapshot found: Listbox Button/Listbox Button/Base: Long Option
[percy] Snapshot found: Listbox Button/Listbox Button/Cascade: RTL
[percy] Snapshot found: Listbox Button/Listbox Button/Cascade: Color
[percy] Snapshot found: Listbox Button/Listbox Button/Cascade: Fluid
[percy] Snapshot found: Listbox Button/Listbox Button/Cascade: Font Size
[percy] Snapshot found: Menu Button/Menu Button/Fake Menu Button: Links
[percy] Snapshot found: Menu Button/Menu Button/Fake Menu Button: Links Disabled
[percy] Snapshot found: Menu Button/Menu Button/Fake Menu Button: Links Fixed Width
[percy] Snapshot found: Menu Button/Menu Button/Fake Menu Button: Links Selected
[percy] Snapshot found: Menu Button/Menu Button: RTL
[percy] Snapshot found: Menu Button/Menu Button: RTL Form
[percy] Snapshot found: Menu Button/Menu Button: Badged
[percy] Snapshot found: Menu Button/Menu Button: Checkbox Items
[percy] Snapshot found: Menu Button/Menu Button: Checkbox Items Disabled Item
[percy] Snapshot found: Menu Button/Menu Button: Collapsed
[percy] Snapshot found: Menu Button/Menu Button: Collapsed Form
[percy] Snapshot found: Menu Button/Menu Button: Expanded
[percy] Snapshot found: Menu Button/Menu Button: Expanded Disabled Item
[percy] Snapshot found: Menu Button/Menu Button: Expanded Form
[percy] Snapshot found: Menu Button/Menu Button: Inherit Colour
[percy] Snapshot found: Menu Button/Menu Button: Inherit Font Size
[percy] Snapshot found: Menu Button/Menu Button: Long Option Text
[percy] Snapshot found: Menu Button/Menu Button: Menu Fix Width
[percy] Snapshot found: Menu Button/Menu Button: Menu Icons
[percy] Snapshot found: Menu Button/Menu Button: Radio Items
[percy] Snapshot found: Menu Button/Menu Button: Radio Items Disabled Item
[percy] Snapshot found: Menu Button/Menu Button: Separator
[percy] Snapshot found: Page Notice/Page Notice: Attention With Button
[percy] Snapshot found: Page Notice/Page Notice: Confirmation With Button
[percy] Snapshot found: Page Notice/Page Notice: General With Button
[percy] Snapshot found: Page Notice/Page Notice: Information With Button
[percy] Snapshot found: Panel Dialog/Panel Dialog: Panel End Secondary Button
[percy] Snapshot found: Section Notice/Section Notice: Attention With Button
[percy] Snapshot found: Section Notice/Section Notice: Confirmation With Button
[percy] Snapshot found: Section Notice/Section Notice: General With Button
[percy] Snapshot found: Section Notice/Section Notice: Information With Button
[percy] Snapshot found: Split Button/Split Button/Base: Fake Primary Collapsed
[percy] Snapshot found: Split Button/Split Button/Base: Fake Primary Expanded
[percy] Snapshot found: Split Button/Split Button/Base: Fake Secondary Collapsed
[percy] Snapshot found: Split Button/Split Button/Base: Fake Secondary Expanded
[percy] Snapshot found: Split Button/Split Button/Base: Fake Tertiary Collapsed
[percy] Snapshot found: Split Button/Split Button/Base: Fake Tertiary Expanded
[percy] Snapshot found: Split Button/Split Button/Base: Primary Collapsed
[percy] Snapshot found: Split Button/Split Button/Base: Primary Expanded
[percy] Snapshot found: Split Button/Split Button/Base: Secondary Collapsed
[percy] Snapshot found: Split Button/Split Button/Base: Secondary Expanded
[percy] Snapshot found: Split Button/Split Button/Base: Tertiary Collapsed
[percy] Snapshot found: Split Button/Split Button/Base: Tertiary Expanded
[percy] Snapshot found: Split Button/Split Button/RTL: Primary Collapsed
[percy] Snapshot found: Split Button/Split Button/RTL: Primary Expanded
[percy] Snapshot found: Split Button/Split Button/RTL: Secondary Collapsed
[percy] Snapshot found: Split Button/Split Button/RTL: Secondary Expanded
[percy] Snapshot found: Split Button/Split Button/RTL: Tertiary Collapsed
[percy] Snapshot found: Split Button/Split Button/RTL: Tertiary Expanded
[percy] Snapshot found: Split Button/Split Button/Size: Long Text Collapsed
[percy] Snapshot found: Split Button/Split Button/Size: Long Text Expanded
[percy] Found 207 snapshots
[percy] Build not created
```

**Comment by ArtBlue** [comment]
> I think the regex might need adjusting. Here is the output from running it on "Button":
> 
> ```
> > @ebay/skin@13.7.1 snapshots:dry
> > npx gulp runSnapshots --dry yes --stories $STORIES "Button"
> 
> [15:30:07] Using gulpfile ~/code/github/ebay/skin/gulpfile.js
> [15:30:07] Starting 'runSnapshots'...
> 
> ************************************************************
> Running Percy Snapshot(s)...
> Snapshot(s):         Button
> Percy Stories Regex: \b(?:Button)\b
> Percy Dry Run:       yes
> Running Snapshot(s)...
> ************************************************************
> 	
> [15:30:07] Finished 'runSnapshots' after 14 ms
> 
> > @ebay/skin@13.7.1 snapshots:execute:dry
> > percy storybook ./docs/storybook --dry-run --include "\b(?:Button)\b"
> 
> [percy] Percy has started!
> [percy] Snapshot found: Button/Button/Base: Busy
> [percy] Snapshot found: Button/Button/Base: Busy Large
> [percy] Snapshot found: Button/Button/Base: Fixed Width
> ```

Yea, I'm aware of the inclusion of all instances of "Button" with that inclusion term. So since this is a regex input, it's very difficult to be selective but yet flexible. I opted for more flexibility instead of being overly selective allowing all iterations of the input to be included. The opposite would be way too selective and might exclude things you may want to see run. That option also would have no way of overriding the stricter regex rule.

The reason I opted for more flexibility is because one can actually override and/or isolate the more narrow set of stories one might want to run. So, if you want to run snapshots for the main buttons, you would do this:

`npm run snapshots:dry 'Button/Button/'`

If you want to run main buttons and menu buttons, you would do this:

`spm run snapshots:dry 'Button/Button/,Menu Button/Menu Button/'`

So on and so forth.

Another alternative I considered was limiting the regex to only look for the patterns at the beginning of the story names but I'm afraid that too might be overly limiting or at the least more verbose to run. In the instance when we make a padding change to a button mixin, for example, which might require ALL button occurrences to have snapshots ran (as in your example), we'd have to include all the button related stories as additional options.

**Comment by agliga** [comment]
> > I think the regex might need adjusting. Here is the output from running it on "Button":
> > ```
> > > @ebay/skin@13.7.1 snapshots:dry
> > > npx gulp runSnapshots --dry yes --stories $STORIES "Button"
> > 
> > [15:30:07] Using gulpfile ~/code/github/ebay/skin/gulpfile.js
> > [15:30:07] Starting 'runSnapshots'...
> > 
> > ************************************************************
> > Running Percy Snapshot(s)...
> > Snapshot(s):         Button
> > Percy Stories Regex: \b(?:Button)\b
> > Percy Dry Run:       yes
> > Running Snapshot(s)...
> > ************************************************************
> > 	
> > [15:30:07] Finished 'runSnapshots' after 14 ms
> > 
> > > @ebay/skin@13.7.1 snapshots:execute:dry
> > > percy storybook ./docs/storybook --dry-run --include "\b(?:Button)\b"
> > 
> > [percy] Percy has started!
> > [percy] Snapshot found: Button/Button/Base: Busy
> > [percy] Snapshot found: Button/Button/Base: Busy Large
> > [percy] Snapshot found: Button/Button/Base: Fixed Width
> > ```
> 
> Yea, I'm aware of the inclusion of all instances of "Button" with that inclusion term. So since this is a regex input, it's very difficult to be selective but yet flexible. I opted for more flexibility instead of being overly selective allowing all iterations of the input to be included. The opposite would be way too selective and might exclude things you may want to see run. That option also would have no way of overriding the stricter regex rule.
> 
> The reason I opted for more flexibility is because one can actually override and/or isolate the more narrow set of stories one might want to run. So, if you want to run snapshots for the main buttons, you would do this:
> 
> `npm run snapshots:dry 'Button/Button/'`
> 
> If you want to run main buttons and menu buttons, you would do this:
> 
> `spm run snapshots:dry 'Button/Button/,Menu Button/Menu Button/'`
> 
> So on and so forth.
> 
> Another alternative I considered was limiting the regex to only look for the patterns at the beginning of the story names but I'm afraid that too might be overly limiting or at the least more verbose to run. In the instance when we make a padding change to a button mixin, for example, which might require ALL button occurrences to have snapshots ran (as in your example), we'd have to include all the button related stories as additional options.

I think I would opt for the option to be stricter matching. If someone is typing in button, they most likely want to run all button stories and not icon-button and such imo. If they want all the stories that match anything with button, then they would have to type out each story. Generally I see that the way people would use this is they change something in a module and they only want to test the modules they touched OR testing everything.

**Comment by agliga** [comment]
I'm getting an error running it locally:
```
(~/build/ebay/skin) % npm run snapshots:dry

> @ebay/skin@13.7.1 snapshots:dry
> npx gulp runSnapshots --dry yes --stories $STORIES

[07:16:09] Using gulpfile ~/build/ebay/skin/gulpfile.js
[07:16:09] Starting 'runSnapshots'...
[07:16:09] 'runSnapshots' errored after 15 ms
[07:16:09] TypeError: stories.split is not a function
    at runSnapshots (/Users/agliga/build/ebay/skin/gulpfile.js:150:30)
    at taskWrapper (/Users/agliga/build/ebay/skin/node_modules/undertaker/lib/set-task.js:13:15)
    at bound (node:domain:421:15)
    at runBound (node:domain:432:12)
    at asyncRunner (/Users/agliga/build/ebay/skin/node_modules/async-done/index.js:55:18)
    at processTicksAndRejections (node:internal/process/task_queues:78:11)
```
I would assume running snapshots dry would run all snapshots.

**Comment by ArtBlue** [comment]
> I'm getting an error running it locally:
> 
> ```
> (~/build/ebay/skin) % npm run snapshots:dry
> 
> > @ebay/skin@13.7.1 snapshots:dry
> > npx gulp runSnapshots --dry yes --stories $STORIES
> 
> [07:16:09] Using gulpfile ~/build/ebay/skin/gulpfile.js
> [07:16:09] Starting 'runSnapshots'...
> [07:16:09] 'runSnapshots' errored after 15 ms
> [07:16:09] TypeError: stories.split is not a function
>     at runSnapshots (/Users/agliga/build/ebay/skin/gulpfile.js:150:30)
>     at taskWrapper (/Users/agliga/build/ebay/skin/node_modules/undertaker/lib/set-task.js:13:15)
>     at bound (node:domain:421:15)
>     at runBound (node:domain:432:12)
>     at asyncRunner (/Users/agliga/build/ebay/skin/node_modules/async-done/index.js:55:18)
>     at processTicksAndRejections (node:internal/process/task_queues:78:11)
> ```
> 
> I would assume running snapshots dry would run all snapshots.
> 
> Adding some documentation on what each command does I think would be helpful.

I was assuming these would be self-explanatory:

```
        "snapshots": "npx gulp runSnapshots --stories $STORIES",
        "snapshots:dry": "npx gulp runSnapshots --dry yes --stories $STORIES",
...
        "snapshots:all": "percy storybook ./docs/storybook --include '.*'",
        "snapshots:all:dry": "percy storybook --dry-run ./docs/storybook --include '.*'",
```

...but sure, I can add some docs. Where would I add those docs?

**Comment by agliga** [comment]
> > I'm getting an error running it locally:
> > ```
> > (~/build/ebay/skin) % npm run snapshots:dry
> > 
> > > @ebay/skin@13.7.1 snapshots:dry
> > > npx gulp runSnapshots --dry yes --stories $STORIES
> > 
> > [07:16:09] Using gulpfile ~/build/ebay/skin/gulpfile.js
> > [07:16:09] Starting 'runSnapshots'...
> > [07:16:09] 'runSnapshots' errored after 15 ms
> > [07:16:09] TypeError: stories.split is not a function
> >     at runSnapshots (/Users/agliga/build/ebay/skin/gulpfile.js:150:30)
> >     at taskWrapper (/Users/agliga/build/ebay/skin/node_modules/undertaker/lib/set-task.js:13:15)
> >     at bound (node:domain:421:15)
> >     at runBound (node:domain:432:12)
> >     at asyncRunner (/Users/agliga/build/ebay/skin/node_modules/async-done/index.js:55:18)
> >     at processTicksAndRejections (node:internal/process/task_queues:78:11)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I would assume running snapshots dry would run all snapshots.
> > Adding some documentation on what each command does I think would be helpful.
> 
> I was assuming these would be self-explanatory:
> 
> ```
>         "snapshots": "npx gulp runSnapshots --stories $STORIES",
>         "snapshots:dry": "npx gulp runSnapshots --dry yes --stories $STORIES",
> ...
>         "snapshots:all": "percy storybook ./docs/storybook --include '.*'",
>         "snapshots:all:dry": "percy storybook --dry-run ./docs/storybook --include '.*'",
> ```
> 
> ...but sure, I can add some docs. Where would I add those docs?

In `CONTRIBUTING.md` there's a percy section. We can add it there.

**Comment by ArtBlue** [comment]
I've updated the regex as requested. Let me know what you want to have done with the docs? Add details for Percy runs? Remove Percy altogether?

**Comment by agliga** [comment]
> I've updated the regex as requested. Let me know what you want to have done with the docs? Add details for Percy runs? Remove Percy altogether?

For me, id like to know what each of those snapshot commands do and when to run them. 

We might consider to add a readme for all the scripts in package.json possibly. There's a lot of commands and it would be good to document them.

**Comment by ianmcburnie** [comment]
I'd say rename the Percy section of CONTRIBUTING.md to "Visual Regression Testing" and add a general overview, plus sub sections for internal contributors and external.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -317,9 +317,57 @@ Every module requires a page in storybook. In addition to the main use cases, tr
...
+
+### Internal Contributors
+
+Internal contributors will need to set up to run Percy snapshots by adding the Percy token to their operating system's environment variables. On a Mac, it would be like so:
+
+`export PERCY_TOKEN=[TOKEN_GOES_HERE]`
+
+This will allow internal contributors to run Percy snapshot tests.
+
+Snapshots will likely be ran by developers locally before pushing up changes. Once they are ran, Percy dashboard should be checked to ensure no unintended style changes have taken place. If there are intended style changes that have occurred, those should be reverted/fixed. Once the set of new local changes is final and in scope with the changes related to the issue, the snapshot run will need to be marked as the canonical version against which future updates are compared.
```
Comment (ianmcburnie) [inline CONTRIBUTING.md:336]: Small typo. I think "If there are intended" should be "If there are unintended"...?

**Review by ianmcburnie — APPROVED** [review]
LGTM! Just one small typo to fix.

**Review by agliga — APPROVED** [review]
Ran it locally with the token. Works like a charm!

File: CONTRIBUTING.md (hunk 1 above)
Comment (ArtBlue) [reply CONTRIBUTING.md:336]: Great catch.

**Pull request merged**